### ISSUE TITLE
Type checking fails in index.js

### ISSUE THREAD

**Issue by ericbn** [sindresorhus/eslint-plugin-unicorn#2394]
Type checking fails in eslint.config.js containing:
```js
// @ts-check
"use strict";
const eslintPluginUnicorn = require("eslint-plugin-unicorn");
const tseslint = require("typescript-eslint");

module.exports = tseslint.config(
	...tseslint.configs.recommended,
	eslintPluginUnicorn.configs["flat/recommended"],
//	^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	{
		rules: {
			"@typescript-eslint/no-var-requires": "off",
			"unicorn/prefer-module": "off",
			"unicorn/prefer-top-level-await": "off",
		},
	},
);
```

First error is:
```
Argument of type '{ rules: any; name: true; plugins: { unicorn: { meta: { name: string; version: string; }; rules: { [x: string]: RuleModule; }; }; }; languageOptions: { globals: any; }; } | { rules: any; plugins: string[]; env: { ...; }; parserOptions: { ...; }; }' is not assignable to parameter of type 'ConfigWithExtends'.
  Type '{ rules: any; name: true; plugins: { unicorn: { meta: { name: string; version: string; }; rules: { [x: string]: Rule.RuleModule; }; }; }; languageOptions: { globals: any; }; }' is not assignable to type 'ConfigWithExtends'.
    Types of property 'name' are incompatible.
      Type 'boolean' is not assignable to type 'string'.ts(2345)
```
which can be fixed by the following change in index.js:
```diff
-const createConfig = (rules, flatConfigName = false) => ({
+const createConfig = (rules, flatConfigName = '') => ({
```

The following error is:
```
Argument of type '{ rules: any; name: string; plugins: { unicorn: { meta: { name: string; version: string; }; rules: { [x: string]: RuleModule; }; }; }; languageOptions: { globals: any; }; } | { rules: any; plugins: string[]; env: { ...; }; parserOptions: { ...; }; }' is not assignable to parameter of type 'ConfigWithExtends'.
  Type '{ rules: any; plugins: string[]; env: { es2024: boolean; }; parserOptions: { ecmaVersion: string; sourceType: string; }; }' is not assignable to type 'ConfigWithExtends'.
    Types of property 'plugins' are incompatible.
      Type 'string[]' is not assignable to type 'Plugins'.
        Index signature for type 'string' is missing in type 'string[]'.ts(2345)
```

Not sure if this is valid change in index.js:
```diff
-			: {...legacyConfigBase, plugins: ['unicorn']}
+			: {...legacyConfigBase, plugins: {unicorn}}
```

**Comment by johanbook** [comment]
I also bumped into this type error when installing the latest of eslint-plugin-unicorn

**Comment by ericbn** [comment]
Looks like this is going to be fixed with the next version, with #2382.
